### PR TITLE
support for wildcards in dispatch patterns

### PR DESCRIPTION
**Pull request by jsheedy**

### CODE REVIEW

File: pythonosc/dispatcher.py [hunk 1]
```diff
@@ -45,7 +45,12 @@ def handlers_for_address(self, address_pattern):
     pattern = pattern + '$'
     pattern = re.compile(pattern)
     matched = [
-        handler for addr, handler in self._map.items() if pattern.match(addr)]
+        handler for addr, handler in self._map.items()
+        if ( pattern.match(addr) # wildcards could be in the incoming pattern
+             # .. or the mapped pattern
+             or (('*' in addr) and re.match(addr.replace('*','[^/]*?/*'), address_pattern))
```
Comment (jsheedy) [inline pythonosc/dispatcher.py]: We could compile this regex just once in Dispatcher.map and store it along with the mapping.

Comment (attwad) [inline pythonosc/dispatcher.py]: no worries, python caches the regexps anyway so let's deal with precompiling if the needs come :)

**Comment by attwad** [comment]
Awesome, thanks for working on this!!

**Comment by wraybowling** [comment]
I was very pleased when I saw this! Thanks, @jsheedy. :D

**Pull request merged**